site/content/entry/citra-progress-report-2017-september.md

```diff
@@ -0,0 +1,360 @@
+load both the base game's and the update's RomFS image, rather than replacing one
+with the other. Games are left to their own devices on how to handle these, and
+so the methods used per game can vary, though they usually just replace changed
+files, picking files from the base game RomFS if they haven't been modified.
+
+Citra, before this PR, had the code for loading games and reading NCCH files all
+mixed into one big piece that fit in with everything else. With this patch,
+[shinyquagsire23](https://github.com/shinyquagsire23) has seperated the loader
+from the NCCH reader, allowing the loader to read multiple NCCHs at once. Additionally,
+whenever a game is loaded, the loader would also check if there is an update title
+installed on Citra's [virtual SD card](/wiki/user-directory/). If there is, it
+would replace the update ExHeader and ExeFS, and load the update RomFS as well.
+Just like a real console!
+
+Most games worked out of the box with updates, and because they wrote the code
+with accuracy in mind, this very same PR has also laid part of the foundation
+needed to handle other features such as DLC support or even using real 3DS SD cards!
+Though, do note that we don't have any estimates on either those or any other
+features, as no one is actively working on either.
+
+## [Implement geometry shader](https://github.com/citra-emu/citra/pull/2865) by [wwylele](https://github.com/wwylele)
+
+The PICA200 GPU has a pipeline similar to [OpenGL's pipeline](https://www.khronos.org/opengl/wiki/Rendering_Pipeline_Overview)
+for rendering 3D objects into a 2D display. I won't go through them all here,
+only the optional geometry shader step. Just after the vertex shader step, if
+enabled, all the vertices are processed by a shader kernel (which is a small
+program that runs directly on a GPU), taking as many vertices as the kernel
+wants as input, and outputting as many vertices as the kernel wants.
+
+Because the kernel in the geometry shader is allowed as many inputs and outputs
+as it wants, it is significantly more powerful and flexible than the vertex shader,
+whose kernel is restricted to only one vertex at a time, both for input and output.
+But for that same reason, geometry shaders are much more complex to program, and
+so many games simply disable it. The games that do not disable it though, tend to
+use it very extensively, to the point of completely breaking graphics if it's not
+implemented.
+
+Multiple uses have been found in the wild for geometry shaders, including but absolutely
+not limited to:
+
+ - Taking one vertex as input, and outputting a rectangle of vertices which can
+ be textured with a sprite. Pok&eacute;mon uses this extensively to render particles
+ whenever a move is used. Monster Hunter takes it a step further and renders
+ *all of its HUD and GUI* with this kernel.
+
+ - Taking a handful of vertices as input, and outputting even more vertices which
```

Comment:
Should we cite any games that do this? Or are we just hypothesizing that some do?

Reply:
@wwylele Confirmed that *specifically* Pokemon and Monster Hunter do these things. Whether *other* games do, I don't know.